app/wasm/test/custom_query_test.go

```diff
@@ -173,6 +173,96 @@ func TestQuerySpotPrice(t *testing.T) {
 	require.InEpsilonf(t, expected+swapFee, price, epsilon, fmt.Sprintf("Outside of tolerance (%f)", epsilon))
 }
 
+func TestQueryEstimatePrice(t *testing.T) {
+	actor := RandomAccountAddress()
+	osmosis, ctx := SetupCustomApp(t, actor)
+	epsilon := 1. // high epsilon (because of rounding?)
```

Comment:
That is too high. The answer could be between 0 and 40 and it would be considered correct.

Please use a normal swap amount, like 1 million (1 OSMO) and then set epsilon to 0.01. If it is more than 1% off we have a serious problem (and should likely be off much less)

Reply:
Ah, I forgot about slippage.
Let's either trade 10.000 (so it is much less than pool funds).
Or multiply the pool funds by 100 so we have small slippage.